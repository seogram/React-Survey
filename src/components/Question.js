import React from 'react'

export const Question = ({ question, Useranswers, step, onAnswerSelected, onSubmit, back, error }) => {

  return (
    <div>
      <h2 className="question">{question.question}</h2>
      <ol>
        {question.answers.map((answer, i) =>
          <li key={`${step}-${i}`} className="answerOption" >
            <input type="radio" name={`question_${step}`} id={`question_${step}_answer_${i}`}
              className='radioCustomButton'
              defaultChecked={i == Useranswers ?
                true : false} value={i} onChange={onAnswerSelected} />
            {' '}
            <label className="radioCustomLabel" htmlFor={`question_${step}_answer_${i}`}>{answer.label}</label>
          </li>
        )}
      </ol>
      <div style={{ textAlign: 'center' }}>
        {step > 0 && <button className='back' onClick={back}>Back</button>}
        <button className='submit' onClick={onSubmit}>Submit</button>
        <p className='error'>{error}</p>
      </div>

    </div>
  )
}

export default Question