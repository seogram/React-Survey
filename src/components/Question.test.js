import React from 'react';
import { shallow } from 'enzyme';
import { Question } from './Question';

describe('Question Component', () => {
    const data =
        {
            "question": "Choose your Favorite Country",
            "answers": [
                {
                    "point": 1,
                    "label": "Germany"
                },
                {
                    "point": 0,
                    "label": "UK"
                },
                {
                    "point": 0,
                    "label": "US"
                },
                {
                    "point": 0,
                    "label": "France"
                }
            ]
        }
    const mockSubmit = jest.fn();
    const mockOnAnswerSelected = jest.fn();
    const mockBack = jest.fn();

    const props = {
        question: data, Useranswers: 1, step: 1,
        onAnswerSelected: mockOnAnswerSelected, onSubmit: mockSubmit,back:mockBack
    };
    const question = shallow(<Question {...props} />);

    it('renders properly', () => {
        expect(question).toMatchSnapshot();
    });

    it('displays the question from props', () => {
        expect(question.find('.question').text()).toEqual('Choose your Favorite Country');
    });

    it('creates a checkboxes in order to choose the answer', () => {
        expect(question.find('.radioCustomButton').exists()).toBe(true);
    });

    it('creates 4 checkboxes ', () => {
        expect(question.find('.radioCustomButton').length).toEqual(4);
    });

});



