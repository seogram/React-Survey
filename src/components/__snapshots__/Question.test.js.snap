// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Component renders properly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h2
        className="question"
    >
        Choose your Favorite Country
    </h2>
    <ol>
        <li
            className="answerOption"
        >
            <input
                className="radioCustomButton"
                defaultChecked={false}
                id="question_1_answer_0"
                name="question_1"
                onChange={[Function]}
                type="radio"
                value={0}
            />
             
            <label
                className="radioCustomLabel"
                htmlFor="question_1_answer_0"
            >
                Germany
            </label>
        </li>
        <li
            className="answerOption"
        >
            <input
                className="radioCustomButton"
                defaultChecked={true}
                id="question_1_answer_1"
                name="question_1"
                onChange={[Function]}
                type="radio"
                value={1}
            />
             
            <label
                className="radioCustomLabel"
                htmlFor="question_1_answer_1"
            >
                UK
            </label>
        </li>
        <li
            className="answerOption"
        >
            <input
                className="radioCustomButton"
                defaultChecked={false}
                id="question_1_answer_2"
                name="question_1"
                onChange={[Function]}
                type="radio"
                value={2}
            />
             
            <label
                className="radioCustomLabel"
                htmlFor="question_1_answer_2"
            >
                US
            </label>
        </li>
        <li
            className="answerOption"
        >
            <input
                className="radioCustomButton"
                defaultChecked={false}
                id="question_1_answer_3"
                name="question_1"
                onChange={[Function]}
                type="radio"
                value={3}
            />
             
            <label
                className="radioCustomLabel"
                htmlFor="question_1_answer_3"
            >
                France
            </label>
        </li>
    </ol>
    <div
        style={
            Object {
                "textAlign": "center",
              }
        }
    >
        <button
            className="back"
            onClick={[Function]}
        >
            Back
        </button>
        <button
            className="submit"
            onClick={[Function]}
        >
            Submit
        </button>
        <p
            className="error"
        />
    </div>
</div>,
  "nodes": Array [
    <div>
      <h2
            className="question"
      >
            Choose your Favorite Country
      </h2>
      <ol>
            <li
                  className="answerOption"
            >
                  <input
                        className="radioCustomButton"
                        defaultChecked={false}
                        id="question_1_answer_0"
                        name="question_1"
                        onChange={[Function]}
                        type="radio"
                        value={0}
                  />
                   
                  <label
                        className="radioCustomLabel"
                        htmlFor="question_1_answer_0"
                  >
                        Germany
                  </label>
            </li>
            <li
                  className="answerOption"
            >
                  <input
                        className="radioCustomButton"
                        defaultChecked={true}
                        id="question_1_answer_1"
                        name="question_1"
                        onChange={[Function]}
                        type="radio"
                        value={1}
                  />
                   
                  <label
                        className="radioCustomLabel"
                        htmlFor="question_1_answer_1"
                  >
                        UK
                  </label>
            </li>
            <li
                  className="answerOption"
            >
                  <input
                        className="radioCustomButton"
                        defaultChecked={false}
                        id="question_1_answer_2"
                        name="question_1"
                        onChange={[Function]}
                        type="radio"
                        value={2}
                  />
                   
                  <label
                        className="radioCustomLabel"
                        htmlFor="question_1_answer_2"
                  >
                        US
                  </label>
            </li>
            <li
                  className="answerOption"
            >
                  <input
                        className="radioCustomButton"
                        defaultChecked={false}
                        id="question_1_answer_3"
                        name="question_1"
                        onChange={[Function]}
                        type="radio"
                        value={3}
                  />
                   
                  <label
                        className="radioCustomLabel"
                        htmlFor="question_1_answer_3"
                  >
                        France
                  </label>
            </li>
      </ol>
      <div
            style={
                  Object {
                        "textAlign": "center",
                      }
            }
      >
            <button
                  className="back"
                  onClick={[Function]}
            >
                  Back
            </button>
            <button
                  className="submit"
                  onClick={[Function]}
            >
                  Submit
            </button>
            <p
                  className="error"
            />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        Useranswers={1}
        back={[Function]}
        onAnswerSelected={[Function]}
        onSubmit={[Function]}
        question={
                Object {
                        "answers": Array [
                          Object {
                            "label": "Germany",
                            "point": 1,
                          },
                          Object {
                            "label": "UK",
                            "point": 0,
                          },
                          Object {
                            "label": "US",
                            "point": 0,
                          },
                          Object {
                            "label": "France",
                            "point": 0,
                          },
                        ],
                        "question": "Choose your Favorite Country",
                      }
        }
        step={1}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "Useranswers": 1,
          "back": [Function],
          "onAnswerSelected": [Function],
          "onSubmit": [Function],
          "question": Object {
            "answers": Array [
              Object {
                "label": "Germany",
                "point": 1,
              },
              Object {
                "label": "UK",
                "point": 0,
              },
              Object {
                "label": "US",
                "point": 0,
              },
              Object {
                "label": "France",
                "point": 0,
              },
            ],
            "question": "Choose your Favorite Country",
          },
          "step": 1,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h2
                    className="question"
          >
                    Choose your Favorite Country
          </h2>
          <ol>
                    <li
                              className="answerOption"
                    >
                              <input
                                        className="radioCustomButton"
                                        defaultChecked={false}
                                        id="question_1_answer_0"
                                        name="question_1"
                                        onChange={[Function]}
                                        type="radio"
                                        value={0}
                              />
                               
                              <label
                                        className="radioCustomLabel"
                                        htmlFor="question_1_answer_0"
                              >
                                        Germany
                              </label>
                    </li>
                    <li
                              className="answerOption"
                    >
                              <input
                                        className="radioCustomButton"
                                        defaultChecked={true}
                                        id="question_1_answer_1"
                                        name="question_1"
                                        onChange={[Function]}
                                        type="radio"
                                        value={1}
                              />
                               
                              <label
                                        className="radioCustomLabel"
                                        htmlFor="question_1_answer_1"
                              >
                                        UK
                              </label>
                    </li>
                    <li
                              className="answerOption"
                    >
                              <input
                                        className="radioCustomButton"
                                        defaultChecked={false}
                                        id="question_1_answer_2"
                                        name="question_1"
                                        onChange={[Function]}
                                        type="radio"
                                        value={2}
                              />
                               
                              <label
                                        className="radioCustomLabel"
                                        htmlFor="question_1_answer_2"
                              >
                                        US
                              </label>
                    </li>
                    <li
                              className="answerOption"
                    >
                              <input
                                        className="radioCustomButton"
                                        defaultChecked={false}
                                        id="question_1_answer_3"
                                        name="question_1"
                                        onChange={[Function]}
                                        type="radio"
                                        value={3}
                              />
                               
                              <label
                                        className="radioCustomLabel"
                                        htmlFor="question_1_answer_3"
                              >
                                        France
                              </label>
                    </li>
          </ol>
          <div
                    style={
                              Object {
                                        "textAlign": "center",
                                      }
                    }
          >
                    <button
                              className="back"
                              onClick={[Function]}
                    >
                              Back
                    </button>
                    <button
                              className="submit"
                              onClick={[Function]}
                    >
                              Submit
                    </button>
                    <p
                              className="error"
                    />
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h2
                    className="question"
          >
                    Choose your Favorite Country
          </h2>
          <ol>
                    <li
                              className="answerOption"
                    >
                              <input
                                        className="radioCustomButton"
                                        defaultChecked={false}
                                        id="question_1_answer_0"
                                        name="question_1"
                                        onChange={[Function]}
                                        type="radio"
                                        value={0}
                              />
                               
                              <label
                                        className="radioCustomLabel"
                                        htmlFor="question_1_answer_0"
                              >
                                        Germany
                              </label>
                    </li>
                    <li
                              className="answerOption"
                    >
                              <input
                                        className="radioCustomButton"
                                        defaultChecked={true}
                                        id="question_1_answer_1"
                                        name="question_1"
                                        onChange={[Function]}
                                        type="radio"
                                        value={1}
                              />
                               
                              <label
                                        className="radioCustomLabel"
                                        htmlFor="question_1_answer_1"
                              >
                                        UK
                              </label>
                    </li>
                    <li
                              className="answerOption"
                    >
                              <input
                                        className="radioCustomButton"
                                        defaultChecked={false}
                                        id="question_1_answer_2"
                                        name="question_1"
                                        onChange={[Function]}
                                        type="radio"
                                        value={2}
                              />
                               
                              <label
                                        className="radioCustomLabel"
                                        htmlFor="question_1_answer_2"
                              >
                                        US
                              </label>
                    </li>
                    <li
                              className="answerOption"
                    >
                              <input
                                        className="radioCustomButton"
                                        defaultChecked={false}
                                        id="question_1_answer_3"
                                        name="question_1"
                                        onChange={[Function]}
                                        type="radio"
                                        value={3}
                              />
                               
                              <label
                                        className="radioCustomLabel"
                                        htmlFor="question_1_answer_3"
                              >
                                        France
                              </label>
                    </li>
          </ol>
          <div
                    style={
                              Object {
                                        "textAlign": "center",
                                      }
                    }
          >
                    <button
                              className="back"
                              onClick={[Function]}
                    >
                              Back
                    </button>
                    <button
                              className="submit"
                              onClick={[Function]}
                    >
                              Submit
                    </button>
                    <p
                              className="error"
                    />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    Useranswers={1}
    back={[Function]}
    onAnswerSelected={[Function]}
    onSubmit={[Function]}
    question={
        Object {
            "answers": Array [
              Object {
                "label": "Germany",
                "point": 1,
              },
              Object {
                "label": "UK",
                "point": 0,
              },
              Object {
                "label": "US",
                "point": 0,
              },
              Object {
                "label": "France",
                "point": 0,
              },
            ],
            "question": "Choose your Favorite Country",
          }
    }
    step={1}
/>,
}
`;
